src/devices/Mbi5027/Mbi5027.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// MBI5027 16-Bit shift register With 3-State output register and error detection.
+    /// Supports SPI and GPIO control.
+    /// </summary>
+    public class Mbi5027 : ShiftRegister
+    {
+        // Datasheet: http://archive.fairchip.com/pdf/MACROBLOCK/MBI5027.pdf
+        private Mbi5027PinMapping _pinMapping;
+
+        /// <summary>
+        /// Initialize a new shift register connected through GPIO.
+        /// </summary>
+        /// <param name="pinMapping">The pin mapping to use by the binding.</param>
+        /// <param name="bitLength">Bit length of register, including chained registers. Default is 16 bits.</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling.</param>
+        /// <param name="shouldDispose">Option (true by default) to dispose the GPIO controller when disposing this instance.</param>
+        public Mbi5027(Mbi5027PinMapping pinMapping, int bitLength = 16, GpioController gpioController = null,  bool shouldDispose = true)
+        : base(new ShiftRegisterPinMapping(pinMapping.Sdi, pinMapping.Clk, pinMapping.LE, pinMapping.OE), bitLength, gpioController, shouldDispose)
+        {
+            _pinMapping = pinMapping;
+            SetupPins();
+        }
+
+        /// <summary>
+        /// Initializes a new shift register device connected through SPI.
+        /// Uses 3 pins (SDI -> SDI, SCLK -> SCLK, CE0 -> LE)
+        /// </summary>
+        /// <param name="spiDevice">SpiDevice used for serial communication.</param>
+        /// <param name="bitLength">Bit length of register, including chained registers. Default is 8 bits.</param>
+        public Mbi5027(SpiDevice spiDevice, int bitLength = 16)
+        : base(spiDevice, bitLength)
+        {
+        }
+
+        /// <summary>
+        /// Enable open/short eror detection mode.
+        /// Requires use of GPIO controller.
+        /// </summary>
+        public void EnableDetectionMode()
+        {
+            if (GpioController is null || _pinMapping.Clk < 0 || _pinMapping.OE < 0 || _pinMapping.LE < 0 || _pinMapping.Sdo < 0)
```

Comment:
I think only -1 should be considered invalid although I'm fine with `<0`

Reply:
Checking for `-1` in particular seems not all that useful. It's not a special number. Why is it different or more likely than `-2`?

Reply:
we picked `-1` by convention but `< 0` is fine as well